### ISSUE TITLE
Remote functions and actors currently cannot close over actor definitions.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#280]
The following currently fails.

```python
import ray

ray.init()

@ray.actor
class Actor(object):
  pass

@ray.remote
def f():
  Actor()
```

Similarly, this example fails.

```python
import ray

ray.init()

@ray.actor
class Actor1(object):
  pass

@ray.actor
class Actor2(object):
  def __init__(self):
    Actor1()

a = Actor2()
```

These examples fail with the following error.

```
PicklingError: Could not pickle object as excessively deep recursion required.
```

This can presumably be fixed in the same way that we handle remote functions closing over remote functions.

**Comment by robertnishihara** [comment]
As a workaround, if the definition of `Actor` is in a different file (say `helper.py`), then the following should work.

```python
import helper

@ray.remote
def f():
  helper.Actor()
```

**Comment by robertnishihara** [comment]
The problem is most likely the call to `super` used inside of the actor decorator.

**Comment by robertnishihara** [comment]
Addressed by #587.
